zapr_noslog_test.go

```diff
@@ -0,0 +1,63 @@
+//go:build !go1.21 && !go1.21
+// +build !go1.21,!go1.21
+
+/*
+Copyright 2023 The logr Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package zapr_test
+
+import (
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// These implementations only exist to allow the tests to compile. Test cases
+// that depend on slog support get skipped at runtime.
+
+func haveSlog() bool {
```

Comment:
```suggestion
func hasSlog() bool {
```
May be a better suited way ?

Reply:
Okay - changed.